sources/lib/api/gcp/data/__init__.py

```diff
@@ -27,10 +27,109 @@ def sql(sql_template, **kwargs):
     sql_template: the template of the SQL statement with named placeholders.
     **kwargs: the dictionary of name/value pairs to use for placeholder values.
   Returns:
-    The formatted SQL statement with placeholders replaced with their values, and
-    an array of Javascript UDFs referenced by the code.
+    The formatted SQL statement with placeholders replaced with their values.
   Raises:
     Exception if a placeholder was found in the SQL statement, but did not have
     a corresponding argument value.
   """
   return _util.Sql.format(sql_template, kwargs)
+
+
+def tokenize(sql):
+  """ This is a crude tokenizer for our limited purposes.
```

Comment:
We need to find the corresponding closing parenthesis for any particular call, so we can wrap the whole thing in (SELECT f1, ... FROM ...). This tokenizer was created primarily for that, but once we were using it for that it made sense to use it to find the function calls as well; the resulting code is quite a bit simpler than an earlier version that used REs to find the calls and the tokenizer to find the call ends.